### PR TITLE
Enabled translations search

### PR DESCRIPTION
**Pull request by thierrymarianne**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Enable searching for translations |
| Type? | improvement |
| Category? | BO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | BOOM-1246 |
| How to test? | Go to International > Translations and search for a translation via the search form |

### CODE REVIEW

**Comment by thierrymarianne** [comment]
Depends on https://github.com/PrestaShop/PrestaShop/pull/6026

File: admin-dev/themes/new-theme/js/translation-page/messages-search.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import Jets from 'jets/jets';
+import $ from 'jquery';
+
+export default function() {
+    $(() => {
+        $('.search-translation form').submit(function () {
+            $('#jetsContent form').addClass('hide');
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/js/translation-page/messages-search.js]: why not use ``hide()|show()`` instead ?

File: admin-dev/themes/new-theme/webpack.config.js [hunk 2]
```diff
@@ -81,19 +81,19 @@ module.exports = {
...
-        booleans: true,
-        if_return: true,
-        join_vars: true,
-        drop_console: true
-      },
-      output: {
-        comments: false
-      }
-    })
+    // new webpack.optimize.UglifyJsPlugin({
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/webpack.config.js]: well , I understand why you did it.

More than ever, we still need a development configuration for webpack

File: src/PrestaShopBundle/Twig/TranslationsExtension.php [hunk 3]
```diff
@@ -149,12 +177,87 @@ public function makeSubtree($tree, $level = 3)
     }
 
     /**
+     * @return array
+     */
+    protected function getSharedEditFormViewProperties()
+    {
+        return array(
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/TranslationsExtension.php]: You probably don't need to pass the locale, right ?

File: src/PrestaShopBundle/Twig/TranslationsExtension.php [hunk 4]
```diff
@@ -175,20 +278,20 @@ public function getName()
...
         }
 
         $output = $this->tagSubject($subject, $level, $id);
 
-        if ($messagesSubtree) {
+        if ($hasMessagesSubtree) {
             $output .= $this->render('button-toggle-messages-visibility.html.twig', array(
-                'label_show_messages' => 'Show messages',
-                'label_hide_messages' => 'Hide messages'
+                'label_show_messages' => $this->translator->trans('Show messages', array(), 'AdminActions', 'en-US'),
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/TranslationsExtension.php]: same apply here

File: src/PrestaShopBundle/Twig/TranslationsExtension.php [hunk 5]
```diff
@@ -198,14 +301,20 @@ protected function concatenateSubtreeHeader($subdomain, $subtree, $level = 2)
         $output .= $this->makeSubtree($subtree, $level + 1);
         $output .= '</div>';
 
-        if ($messagesSubtree) {
+        if ($hasMessagesSubtree) {
             $output .= $this->render('button-go-to-pagination-bar.html.twig', array(
                 'domain' => $id,
-                'label' => 'Go to previous navigation menu',
+                'label' => $this->translator->trans('Go to previous navigation menu', array(), 'AdminActions', 'en-US'),
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Twig/TranslationsExtension.php]: and here (as a reminder)

File: admin-dev/themes/new-theme/webpack.config.js (hunk 2 above)
Comment (thierrymarianne) [inline admin-dev/themes/new-theme/webpack.config.js]: Sorry, restoring it right now :)

File: admin-dev/themes/new-theme/js/translation-page/messages-search.js (hunk 1 above)
Comment (thierrymarianne) [inline admin-dev/themes/new-theme/js/translation-page/messages-search.js]: When inspecting the dom, I find it more obvious with classes which element should be visible or not without having to inspect element by element. Moreover with classes one can tell how many items are hidden in console using the appropiate class selector. It makes testing easier.

File: src/PrestaShopBundle/Twig/TranslationsExtension.php (hunk 3 above)
Comment (thierrymarianne) [inline src/PrestaShopBundle/Twig/TranslationsExtension.php]: Right!

File: src/PrestaShopBundle/Twig/TranslationsExtension.php (hunk 5 above)
Comment (thierrymarianne) [inline src/PrestaShopBundle/Twig/TranslationsExtension.php]: Thanks

**Comment by vincentbz** [comment]
Tested and ok, thanks !

**Comment by mickaelandrieu** [comment]
Thank you @thierrymarianne :)

**Pull request merged**